mempool/cat/reactor.go

```diff
@@ -0,0 +1,443 @@
+package cat
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+
+	cfg "github.com/tendermint/tendermint/config"
+	"github.com/tendermint/tendermint/crypto/tmhash"
+	"github.com/tendermint/tendermint/libs/log"
+	"github.com/tendermint/tendermint/mempool"
+	"github.com/tendermint/tendermint/p2p"
+	protomem "github.com/tendermint/tendermint/proto/tendermint/mempool"
+	"github.com/tendermint/tendermint/types"
+)
+
+const (
+	// default duration to wait before considering a peer non-responsive
+	// and searching for the tx from a new peer
+	defaultGossipDelay = 200 * time.Millisecond
+
+	// Content Addressable Tx Pool gossips state based messages (SeenTx and WantTx) on a separate channel
+	// for cross compatibility
+	MempoolStateChannel = byte(0x31)
+
+	// tx_key + node_id + buffer (for proto encoding)
```

Comment:
is the `node_id` included in the `tmhash.Size` or is 10 equal to `node_id + buffer`?

can be make a constant out of 10 and explain the chosen size?